vi/news/_posts/2015-04-13-ruby-openssl-hostname-matching-vulnerability.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: news_post
+title: "CVE-2015-1855: Xác nhận Hostname Ruby OpenSSL"
+author: "zzak"
+translator: "Dôn Đinh"
+tags: security
+date: 2015-04-13 12:00:00 +0000
+lang: vi
+---
+
+Tiện ích OpenSSL của Ruby có một nhược điểm khi có quá nhiều các hostnames tương ứng, có thể dẫn đến lỗi tương tự như [CVE-2014-1492][CVE-2014-1492]. Các lỗi tương tự như vậy cũng được tìm thấy trong [Python][python-hostname-bug].
+
+Lỗ hỗng này đã được xác định trong [CVE-2015-1855][CVE-2015-1855].
+
+Đó là lý do các bạn nên nâng cấp Ruby.
+
+## Chi tiết
+
+Sau khi xem qua [RFC 6125][RFC-6125] và [RFC 5280][RFC-5280], chúng tôi đã tìm thấy nhiều vi phạm của các hostname tương ứng, đặc biệt là xác nhận các ký tự đại diện.
+
+Hiện tại, tiện ích OpenSSL của Rubys sẽ cung cấp một chuỗi dựa trên các thuộc toán tương ứng, nghiêm ngặt dựa theo các khuyến cáo của các RFC. Đặc biệt, việc kết hợp của nhiều hơn một ký tự đại diện cho mỗi vấn đề/SAN không còn được cho phép. Đồng thời, bây giờ việc so sánh các giá trị sẽ không còn nhạy cảm.
+
+Sự thay đổi này sẽ có ảnh hưởng đến `OpenSSL::SSL#verify_certificate_identity` của Ruby.
+
+Cụ thể ở đây:
+
+* Chỉ một ký tự đại diện trong phần bên trái của hostname được cho phép.
+* Hiện tại, các tên IDNA chỉ tương ứng với một ký tự đại diện đơn giản (ví dụ '*.domain').
+* Vấn đề/SAN sẽ được giới hạn chỉ là ký tự ASCII.
+
+Tất cả mọi người đang hoạt động trên các phiên bản bị ảnh hưởng này nên nâng cấp ngay.
+
+## Các phiên bản bị ảnh hưởng
+
+* Tất cả phiên bản đầu tiên của Ruby 2.0 đến bản vá 2.0.0 p645
+* Tất cả phiên bản đầu tiên của Ruby 2.1 đến Ruby 2.1.6
+* Tất cả phiên bản đầu tiên của Ruby 2.2 đến Ruby 2.2.2
+* Sửa đổi đầu tiên 50292
+
+## Credits
```

Comment:
I'll show you :) Please take a look
![vietgle_tra_t _-_d nh_nghia_c a_t __credit__trong_t _di n_l c_vi t](https://cloud.githubusercontent.com/assets/6791942/7216262/6a339046-e623-11e4-81c3-7471915aad41.png)